lib/app/features/feed/create_article/views/pages/topic_select_modal/topic_select_modal.dart

```diff
@@ -0,0 +1,120 @@
+// SPDX-License-Identifier: ice License 1.0
+
+import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:ion/app/components/inputs/search_input/search_input.dart';
+import 'package:ion/app/components/screen_offset/screen_side_offset.dart';
+import 'package:ion/app/extensions/extensions.dart';
+import 'package:ion/app/features/feed/data/models/topic_type.dart';
+import 'package:ion/app/features/feed/providers/article/select_topics_provider.dart';
+import 'package:ion/app/router/components/navigation_app_bar/navigation_app_bar.dart';
+import 'package:ion/generated/assets.gen.dart';
+
+class TopicSelectModal extends HookConsumerWidget {
+  const TopicSelectModal({
+    super.key,
+  });
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final selectedTopics = ref.watch(selectTopicsProvider);
+    final colors = context.theme.appColors;
+    final textStyles = context.theme.appTextThemes;
+    final searchValue = useState('');
+
+    final filteredTopics = useMemoized(
+      () {
+        final query = searchValue.value.toLowerCase();
+        return TopicType.values.where((topic) {
+          final title = topic.getTitle(context).toLowerCase();
+          return title.contains(query);
+        }).toList();
+      },
+      [searchValue.value],
+    );
+
+    void toggleTopicSelection(TopicType topic) {
```

Comment:
Can you consider using the `useSelectedState` hook instead of this method and without `selectTopicsProvider`?